graphiti_core/helpers.py

```
@@ -99,3 +104,15 @@ async def _wrap_coroutine(coroutine):
             return await coroutine
 
     return await asyncio.gather(*(_wrap_coroutine(coroutine) for coroutine in coroutines))
+
+def handle_datatime_objects(datetime_obj: Any) -> datetime | None:
+    
+    if isinstance(datetime_obj, str):
+        try:
+            time_value = datetime.fromisoformat(datetime_obj)
+        except ValueError:
+            raise ValueError(f"Invalid date format: {datetime_obj}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)